src/main/java/com/google/sps/data/Places.java

```diff
@@ -142,17 +146,78 @@
     return stream;
   }
 
+  /**
+   * This function getHiddenGems converts the Set<PlaceSearchResult> of hidden places
+   * to the type Set<HiddenGem> of hidden gems. It finds additional relevant
+   * information on the hidden gems using the Place's API.
+   * @param places           The Set of PlaceSearchResults of hidden gems to be converted.
+   * @return Set<HiddenGem>  This returns a set of Hidden Gems.
+   */
+  public static Set<HiddenGem> convertToHiddenGems(Set<PlacesSearchResult> places) {
+    GeoApiContext context = new GeoApiContext.Builder()
+      .apiKey(GetConfigProperties.getApiKey())
+      .build();
+
+    return fetchHiddenGemsFromApi(context, places);
+  }
+
+  /**
+   * This converts a set of places into a set of hidden gems. 
+   * Iterate over the set of places and find the additional details, then create
+   * the hidden gem object by populating all the relevant data in the constructor.
+   * @param context           The GeoApiContext to be used for the Place Details Request
+   * @param places            The set of PlaceSearch results to be converted.
+   * @return Set<HiddenGem>   This returns a set of Hidden Gems with all information populated
+   *                          from the place search and place details results. 
+   */
+  public static Set<HiddenGem> fetchHiddenGemsFromApi(GeoApiContext context, Set<PlacesSearchResult> places) {
+
+    Set<HiddenGem> hiddenGems = new HashSet<>();
+
+    for(PlacesSearchResult place : places){
+      try {
+        PlaceDetails placeDetails = PlacesApi.placeDetails(context, place.placeId)
+        .fields(
+            PlaceDetailsRequest.FieldMask.FORMATTED_ADDRESS,
+            PlaceDetailsRequest.FieldMask.PRICE_LEVEL,
+            PlaceDetailsRequest.FieldMask.WEBSITE)
+        .await();
+
+        hiddenGems.add(new HiddenGem(
```

Comment:
just for what it's worth, this is why I was suggesting having a second constructor then you could just do new HiddenGem(place, placeDetails);